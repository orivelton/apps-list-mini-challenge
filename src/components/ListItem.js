import React from 'react'

const ListItem = ({
  name,
  description,
  categories,
  subscriptions,
  setSearchTerm,
  setIsCategories
}) => (
  <li>
    <div className="app-item">
      <div className="box-info">
        <div className="box-info--content">
          <div className="description">
            <h1>{name}</h1>
            <p>{description}</p>
          </div>
          <div className="tags">
            { categories && categories.map(item => <span onClick={() => { setSearchTerm(item); setIsCategories(true)}} key={item}>{item} {'/'}</span>) }
          </div>
        </div>
        <div className="box-info--footer">
          <ul>
            {
              subscriptions && subscriptions.map(({name, price}) => 
                <li key={name}>
                  <span>{name}</span>
                  <h3>
                    {price}<sup>â‚¬</sup>
                  </h3>
                </li>
              )
            }
          </ul>
        </div>
      </div>
    </div>
  </li>
);

export default ListItem;
