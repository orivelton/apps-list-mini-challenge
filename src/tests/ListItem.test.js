import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../components/ListItem';

test('renders correctly', () => {
  const tree = renderer.create(<ListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
