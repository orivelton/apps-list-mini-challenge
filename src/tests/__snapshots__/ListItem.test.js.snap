// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <div
    className="app-item"
  >
    <div
      className="box-info"
    >
      <div
        className="box-info--content"
      >
        <div
          className="description"
        >
          <h1 />
          <p />
        </div>
        <div
          className="tags"
        />
      </div>
      <div
        className="box-info--footer"
      >
        <ul />
      </div>
    </div>
  </div>
</li>
`;
